### PR TITLE
replace generic method CtElement.replace() by more specifics methods …

### PR DESCRIPTION
**Pull request by petitpre**
…in CtStatement, CtExpression, etc

### CODE REVIEW

File: src/main/java/spoon/reflect/declaration/CtField.java [hunk 1]
```diff
@@ -35,4 +35,9 @@
 	 * @see spoon.reflect.declaration.CtNamedElement#getReference()
 	 */
 	CtFieldReference<T> getReference();
+
+	/**
+	 * Replaces this element by another one.
+	 */
+	<R extends T> void replace(CtField<R> element);
```
Comment (GerardPaligot) [inline src/main/java/spoon/reflect/declaration/CtField.java]: I'm not sure it is relevant to replace a field.

File: src/main/java/spoon/support/reflect/declaration/CtElementImpl.java [hunk 2]
```diff
@@ -328,14 +328,14 @@ private boolean compare(Object o1, Object o2) {
...
-		}
-	}
+//	public void replace(Filter<? extends CtElement> replacementPoints,
+//			CtElement element) {
+//		List<? extends CtElement> l = Query
+//				.getElements(this, replacementPoints);
+//		for (CtElement e : l) {
+//			e.replace(element);
+//		}
+//	}
```
Comment (GerardPaligot) [inline src/main/java/spoon/support/reflect/declaration/CtElementImpl.java]: If it is no more used, remove this method. We have the git history if we would like to retrieve something.

File: src/main/java/spoon/reflect/declaration/CtField.java (hunk 1 above)
Comment (monperrus) [inline src/main/java/spoon/reflect/declaration/CtField.java]: It may be useful in some cases. We may add a precondition on the name and/or the type.

Comment (monperrus) [inline src/main/java/spoon/reflect/declaration/CtField.java]: To be consistent, we should add CtMethod.replace

File: src/main/java/spoon/reflect/code/CtExpression.java [hunk 3]
```diff
@@ -46,5 +46,10 @@
 	 * Adds a type cast.
 	 */
 	void addTypeCast(CtTypeReference<?> type);
+
+	/**
+	 * Replaces this element by another one.
+	 */
+	<E extends T> void replace(CtExpression<E> element);
```
Comment (monperrus) [inline src/main/java/spoon/reflect/code/CtExpression.java]: add a specific test case for CtExpression.replace?

File: src/main/java/spoon/reflect/code/CtStatement.java [hunk 4]
```diff
@@ -64,4 +64,8 @@ void insertBefore(CtStatementList statements)
 	 */
 	void setLabel(String label);
 
+	/**
+	 * Replaces this element by another one.
+	 */
+	void replace(CtStatement element);
```
Comment (monperrus) [inline src/main/java/spoon/reflect/code/CtStatement.java]: add a specific test case for CtStatement.replace?

**Pull request merged**